common/curl-p/ptrit.c

```diff
@@ -0,0 +1,75 @@
+/*
+ * (c) 2016 Paul Handy, based on code from come-from-beyond
+ */
+
+#include <stdlib.h>
+#include <string.h>
+
+#include "indices.h"
+#include "ptrit.h"
+
+static const size_t CURL_INDEX[STATE_LENGTH + 1] = {__INDEX_TABLE};
+
+static void ptrit_sbox(ptrit_t *const c, ptrit_t const *const s) {
+  ptrit_s alpha, beta, gamma, delta;
+  size_t i = 0;
+
+  for (; i < STATE_LENGTH; ++i) {
+    alpha = s[CURL_INDEX[i]].low;
+    beta = s[CURL_INDEX[i]].high;
+    gamma = s[CURL_INDEX[i + 1]].high;
+    delta = (alpha | (~gamma)) & (s[CURL_INDEX[i + 1]].low ^ beta);
+
+    c[i].low = ~delta;
+    c[i].high = (alpha ^ gamma) | delta;
+  }
+}
+
+void ptrit_curl_init(PCurl *const ctx, CurlType type) {
+  ptrit_curl_reset(ctx);
+  ctx->type = type;
+}
+
+void ptrit_curl_absorb(PCurl *const ctx, ptrit_t const *const trits,
+                       size_t length) {
+  size_t numChunks =
+      length / HASH_LENGTH_TRIT + ((length % HASH_LENGTH_TRIT) ? 1 : 0);
```

Comment:
How about change from `(length % HASH_LENGTH_TRIT) ? 1 : 0` to `!!(length % HASH_LENGTH_TRIT)`?